### ISSUE TITLE
Improving isocurve calculation speed (Isocurve Visual)

### ISSUE THREAD

**Issue by kmuehlbauer** [vispy/vispy#989]
I use this example with an added 'auto' timer and numpy roll the data between the timer updates to create some motion:

https://github.com/vispy/vispy/blob/master/examples/basics/scene/isocurve.py

For my system I get frame rates of around 12 fps using the original vispy isocurve calculation (in geometry). 

After that I tried the pure python scikit-image implementation of contour finding: http://scikit-image.org/docs/dev/auto_examples/plot_contours.html by simply exchanging the calculation procedure within the Isocurve Visual draw routine:

```
# new import
from skimage import measure
# exchange calculation function
# paths = isocurve(self._data.astype(float).T, self._level, extend_to_edge=True, connected=True)
paths = measure.find_contours(self._data.astype(float).T, self._level)
```
This improves fps to around 40.

After that I tried the matplotlib implementation (C module) of contour finding, which is shown here: http://stackoverflow.com/questions/18304722/python-find-contour-lines-from-matplotlib-pyplot-contour

```
# new import
from matplotlib import _cntr as cntr
# exchange calculation function
# paths = isocurve(self._data.astype(float).T, self._level, extend_to_edge=True, connected=True)
X,Y = np.meshgrid(np.arange(self._data.shape[0]),np.arange(self._data.shape[1]))
c = cntr.Cntr(X,Y,self._data.astype(float))
nlist = c.trace(self._level, self._level, 0)
paths = nlist[:len(nlist)//2]
```
This improves the fps to 60.

What steps are necessary to implement one (or both) of the faster isocurve routines? 

It would also be `nice to have` a Collection of Isocurves  where you can create several Isocurves at once by handing over an array of levels.

**Comment by campagnola** [comment]
As a first step, we could attempt to import these routines first from matplotlib and then scikit-image and then fail over to the built in version if those packages are not available.

**Comment by kmuehlbauer** [comment]
resolved by campagnola/vispy#12
